stable/prometheus/values.yaml

```diff
@@ -419,6 +422,9 @@ alertmanagerFiles:
 ## Prometheus server ConfigMap entries
 ##
 serverFiles:
+  # ConfigMap override where fullname is {{.Release.Name}}-{{.Values.serverFiles.configMapOverrideName}}
+  configMapOverrideName: ""
```

Comment:
Let's move this up to the `server` section. Also, wrap the contents of `templates/server-configmap.yaml` in a conditional so that if `server.configMapOverrideName` is set, we don't create an unnecessary resource. Then add a note here (and in the README) to that effect.